airflow/executors/celery_executor.py

```diff
@@ -252,10 +253,25 @@ def __init__(self):
         )
         self.task_publish_retries: Counter[TaskInstanceKey] = Counter()
         self.task_publish_max_retries = conf.getint("celery", "task_publish_max_retries", fallback=3)
+        self.task_stuck_in_queued_check_interval = conf.getint("celery",
+                                                               "task_stuck_in_queued_check_interval",
+                                                               fallback=None)
+        self.task_stuck_in_queued_timeout = conf.getint("celery",
+                                                        "task_stuck_in_queued_timeout",
+                                                        fallback=300)
+        self.event_scheduler: EventScheduler | None = None
 
     def start(self) -> None:
         self.log.debug("Starting Celery Executor using %s processes for syncing", self._sync_parallelism)
 
+        if self.task_stuck_in_queued_check_interval:
+            self.event_scheduler = EventScheduler()
+
+            self.event_scheduler.call_regular_interval(
+                self.task_stuck_in_queued_check_interval,
+                self.clear_not_launched_queued_tasks,
+            )
```

Comment:
I don’t think we need an extra scheduler for this, but can probably check during a scheduler loop iteration. Probably during `_critical_section_enqueue_task_instances` to avoid race conditions (a scheduler tries to schedule a task that’s cleared in another).